processor/src/main/resources/org/mapstruct/ap/internal/model/assignment/SetterWrapperForCollectionsAndMaps.ftl

```diff
@@ -43,20 +43,16 @@
 -->
 <#macro callTargetWriteAccessor>
     <@lib.handleNullCheck>
-      <#if directAssignment>
-        ${ext.targetBeanName}.${ext.targetWriteAccessorName}( <@wrapLocalVarInCollectionInitializer/> );
-    <#else>
-        ${ext.targetBeanName}.${ext.targetWriteAccessorName}( ${localVarName} );
-    </#if>
```

Comment:
Hey @filiphr ,

I tried your suggestion.. But there's a side effect. The compress also acts on the call site. So the result is (see Issue289 for instance):

```java
        Collection<TargetElement> collection = sourceElementCollectionToTargetElementCollection( source.getCollection() ); if ( collection != null ) { targetWithSetter.setCollection( collection ); }        
```
In one line. Perhaps an error in Freemarker..